When exporting detections to csv also add option to only export selected types

f4str opened the pull request:
fixes #695 

Web version new checkbox: 
![image](https://user-images.githubusercontent.com/30541973/120842749-1e7a3580-c53b-11eb-8596-9d069464fb21.png)

Desktop version new checkbox:
![image](https://user-images.githubusercontent.com/30541973/120842889-60a37700-c53b-11eb-9aea-c4324827d21a.png)

BryonLewis commented on client/platform/web-girder/views/Export.vue:
```diff
@@ -34,15 +34,18 @@ export default defineComponent({
     if (props.blockOnUnsaved) {
       save = useHandler().save;
       pendingSaveCount = usePendingSaveCount();
+      checkedTypes = useCheckedTypes();
```
I don't know if this should be in the `blockOnUnsaved` prop and maybe it should be outside of it where the variable is initialized?  I know `blockOnUnsaved` set outside of here in the `ViewLoader.vue` but the default properly is false.  `checkedTypes` once bound in the beginning should auto update anytime the checked types are changed.  I could be mistaken on this and it's handling an edge case I don't know about.

subdavis replied:
`blockOnUnsaved` is a proxy indicator for "this export dialog was mounted on the viewer instead of the data browser".

If this were outside this block, it would cause a runtime exception because the Provides api was not injected.

There's a conditional on menuOpen below to prevent requests to the server unless the menu is actually open.

No changes needed, IMO.

BryonLewis replied:
thanks, that was the case I was missing.

BryonLewis commented on server/dive_server/serializers/viame.py:
```diff
@@ -280,17 +280,20 @@ def export_tracks_as_csv(
     filenames=None,
     fps=None,
     header=True,
+    typeFilter=set(),
```
Might be an opportunity to add some more dataset export testing to `test_serialize_viame_csv.py`?  I'd check with @subdavis and see what he thinks about possible adding the typeFiltering to here.  Could also be an indicator that we start adding nodejs serializer testing to match the python.  Ensuring that same data in = same data out between the two.

subdavis replied:
Agree with Bryon

BryonLewis reviewed (commented):
In my testing this works really well on desktop/web.  I just have one general question about the placement of `useCheckedTypes` in `Export.vue` (I could very well be missing something).  Other than that the possibility of maybe adding some additional testing the python export script and maybe some initial testing to the node.js/desktop side.

subdavis commented on client/platform/desktop/frontend/components/Export.vue:
```diff
@@ -183,6 +187,16 @@ export default defineComponent({
+          <template v-if="checkedTypes.length">
+            <v-checkbox
+              v-model="data.excludeUncheckedTypes"
+              label="export checked types only"
```
Maybe add a `details` prop with `Export only the types currently enabled in the type filter`.  Get rid of `hide-details`.

I think the extra explanation is needed.

subdavis reviewed (changes requested):
* Agree with bryon about adding a test case or two in python
* Couple other minor comments.

Looks great, confirmed working, no issues.

f4str commented:
Added some more unit tests for exporting with type filters and importing with missing types. Also added the checkbox hint.
![image](https://user-images.githubusercontent.com/30541973/121241946-fe63b280-c869-11eb-9f2c-465edbaf3583.png)

subdavis reviewed (approved):
:100: looks great, thanks.

I noticed in the unit tests that we aren't testing the confidence threshold either.  It would be good to test that, especially in the case where both confidence and track type are filtered and a track has multiple types where the behavior might conflict.
